### PR TITLE
chore: update triplewz/poseidon

### PR DESCRIPTION
**Pull request by magik6k**
## Related Issues
https://github.com/triplewz/poseidon/pull/2

## Proposed Changes
Curio pulls in the new version of the Poseidon library which switched to generic fr.Element. We need that to support SnapDeal decode in pure Go, and can't use the older version.

## Checklist

Before you mark the PR ready for review, please make sure that:

- [x] Commits have a clear commit message.
- [x] PR title conforms with [contribution conventions](https://github.com/filecoin-project/lotus/blob/master/CONTRIBUTING.md#pr-title-conventions)
- [x] Update CHANGELOG.md or signal that this change does not need it per [contribution conventions](https://github.com/filecoin-project/lotus/blob/master/CONTRIBUTING.md#changelog-management)
- [x] New features have usage guidelines and / or documentation updates in
  - [x] [Lotus Documentation](https://lotus.filecoin.io)
  - [x] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [x] Tests exist for new functionality or change in behavior
- [x] CI is green

### CODE REVIEW

File: go.mod
Comment (Stebalien) [inline go.mod:136]: Any chance we could ask for an upstream release?

File: storage/sealer/commitment/commr.go
Comment (Stebalien) [inline storage/sealer/commitment/commr.go:28]: Confirming that there _is_ a golden test for this here: https://github.com/filecoin-project/lotus/blob/f6d4fc7ccb0a53fd49cf01a82ab3a9528345afab/storage/sealer/commitment/commr_test.go#L9-L35.

**Review by Stebalien — APPROVED** [review]
I'd prefer a released version, but this seems reasonable. From what I can tell, this is _only_ used in sealing, we have a separate flow to _check_ the proofs. That means this change should be covered by the itests.

But just to double check: @magik6k have you tested the lotus-miner sealing flow and/or confirmed that it's tested (enough) in CI? I now some proof types aren't generally tested because they're too slow to test in CI.

File: go.mod
Comment (magik6k) [reply go.mod:136]: Not sure how to get to the maintainer, @vmx might know though.

File: storage/sealer/commitment/commr.go
Comment (magik6k) [reply storage/sealer/commitment/commr.go:28]: Correct, that test is exactly that

File: go.mod
Comment (magik6k) [reply go.mod:136]: (I wouldn't block this PR on the upstream tag, this is actively blocking us on updating Curio to latest NV, and Curio is now most of calibnet)

**Comment by magik6k** [comment]
> From what I can tell, this is only used in sealing, we have a separate flow to check the proofs. That means this change should be covered by the itests.

It's only used with "External PC2 binary" mode of sealing, which IIUC isn't adopted pretty much anywhere (that was used for brownfield supraseal pc2, which while faster wasn't very stable). We also use the new version of this library in Curio for many other parts of the proving system, and it definitely works fine there.

**Comment by rjan90** [comment]
> It's only used with "External PC2 binary" mode of sealing, which IIUC isn't adopted pretty much anywhere (that was used for brownfield supraseal pc2, which while faster wasn't very stable).

Can confirm that the "External PC2 binary" mode of sealing is not well adopted, because it has the downside of only working with CC-sectors currently: and switching between external PC2 binary / internal PC modes depending if it is data or not in the sectors, would be a very painful experience.

File: go.mod
Comment (Stebalien) [reply go.mod:136]: I've filed an issue, no reason to wait.

**Review by Stebalien — APPROVED** [review]

**Pull request merged**
